### ISSUE TITLE
Build a pick up object functionality. 

### ISSUE THREAD

**Issue by RMSteenwijk** [Nexus-Aurora/vr-unity-demo#9]
For example the half life pick up system that a lot of games use

Consider the different controls, note for now we just support:
PC,
VR

**Comment by DeadlyMagikarps** [comment]
That's what I assume. It's a pain in the butt system to make yourself. But it is a very rewarding system. I've only seen tutorials for Unreal. Haven't found one for Unity. It would work the same way, but implementation would be much more hands one.

**Comment by brycekahle** [comment]
#19 has a very basic version of this where you can point at an object, press grip button, and it flies into your hand.

**Comment by RMSteenwijk** [comment]
Yes, and the fps controls for pc players working like half-life 2 with the pick up and rotate mechanics

**Comment by PranavKishan213** [comment]
https://circuitstream.com/blog/grab-oculus/

Have you guys considered this tutorial?

**Comment by PranavKishan213** [comment]
https://circuitstream.com/blog/grab-vive/

One for the HTC Give.

**Comment by PranavKishan213** [comment]
https://youtu.be/w5y8iukw6yY

And I found this one on a Unity forum.

**Comment by DeadlyMagikarps** [comment]
> https://circuitstream.com/blog/grab-vive/
> 
> One for the HTC Vive.



> https://circuitstream.com/blog/grab-oculus/
> 
> Have you guys considered this tutorial?

The only problem with these are they are controller specific. I don't think they work with WMR controllers or Index controllers. Standalone headsets probably don't work either. 

If we end up moving away from steamvr and only using unitys OpenVR/XR system, then the definitely won't work. 

SteamVR already provides built in grabbing support so honestly, don't even know if the above is needed. The only thing that would need to be altered is the half life style where it like magnetizes to your hand when picking up.

**Comment by RMSteenwijk** [comment]
@PranavKishan213 Please stop spamming suggestions and try and build them to see if they are actually worthwhile

**Comment by PranavKishan213** [comment]
Alright. I didn't think that would be spam.

**Comment by olawlor** [comment]
One cross-platform way to do pickup would be to hook a trigger box up to a very short script that moves a GameObject into Player/Carried (an existing Empty in the Unity object hierarchy, at least in the URP branch).  This would let us make spawners like water purifiers that just create water objects that players can walk away with; if there are tons of objects scattered around you clearly don't want to pick them each up by walking over them!

**Comment by DeadlyMagikarps** [comment]
> One cross-platform way to do pickup would be to hook a trigger box up to a very short script that moves a GameObject into Player/Carried (an existing Empty in the Unity object hierarchy, at least in the URP branch). This would let us make spawners like water purifiers that just create water objects that players can walk away with; if there are tons of objects scattered around you clearly don't want to pick them each up by walking over them!

This can all be done using tags! If we have an array of list or something like that and just check for specific tags, the trigger can be activated correctly. In the pickup code, just use a switch case statement and do accordingly to what the player walked over.
